tg/tg_eval.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2024 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package tg
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gruntwork-io/go-commons/errors"
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/util"
+	"github.com/terramate-io/terramate/project"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+)
+
+// Find a parent Terragrunt configuration file in the parent folders above the current Terragrunt configuration file
+// and return its path
+func findInParentFolders(
+	ctx *config.ParsingContext,
+	rootdir string,
+	mod *Module,
+	params []string,
+) (abspath string, err error) {
+	const (
+		noMatchedPats = 1
+		matchedPats   = 2
+	)
+
+	defer func() {
+		if err == nil {
+			mod.TriggerPaths = append(mod.TriggerPaths, project.PrjAbsPath(rootdir, abspath))
+		}
+	}()
```

Comment:
hack: tracks file lookups.

I don't know if we should track this even if the actual returned file is not used.